recipe_scrapers/bergamot.py

```diff
@@ -0,0 +1,84 @@
+# mypy: allow-untyped-defs
+import requests
+
+from ._abstract import HEADERS, AbstractScraper
+from ._utils import url_path_to_dict
+
+
+class Bergamot(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+
+        url_dict = url_path_to_dict(url)
+        path = url_dict.get("path")
+        recipe_id = path.split("/")[-1]
+
+        data_url = f"https://api.bergamot.app/recipes/shared?r={recipe_id}"
+        response = requests.get(
+            data_url, headers=HEADERS, proxies=proxies, timeout=timeout
+        )
+        self.data = response.json()
+
+    @classmethod
+    def host(cls):
+        return "dashboard.bergamot.app"
+
+    def canonical_url(self):
+        source_url = self.data.get("sourceUrl")
+        return source_url if source_url else self.url
```

Comment:
This _could_ be reduced down to:

```suggestion
        return self.data.get("sourceUrl") or self.url
```

But let me know what you think about that - I'm not sure whether it's any better or easier to understand (maybe more difficult).